packages/vue-components/src/annotations/AnnotatePoint.vue

```diff
@@ -198,13 +198,21 @@ export default {
       return (this.hasContent || this.hasHeader) && (this.legend === 'popover' || this.legend === 'both');
     },
     computedBottomHeader() {
-      if (this.label !== '' && this.header === '') {
-        return this.label;
+      const labelSlotContent = this.$scopedSlots.label === undefined ? undefined : this.$scopedSlots.label();
+      const headerSlotContent = this.$scopedSlots.header === undefined
+        ? undefined : this.$scopedSlots.header();
+
+      const labelText = labelSlotContent === undefined ? undefined : labelSlotContent[0].children[0].text;
+      const labelHeader = headerSlotContent === undefined ? undefined : headerSlotContent[0].children[0].text;
```

Comment:
We can use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) here, i.e.

```suggestion
      const labelSlotContent = this.$scopedSlots.label?.();
      const headerSlotContent = this.$scopedSlots.header?.();

      const labelText = labelSlotContent?.[0]?.children?.[0]?.text;
      const labelHeader = headerSlotContent?.[0]?.children?.[0]?.text;
```

You can probably combine these to compact them even further but I think this is fine enough considering readability.

Reply:
If you have further thoughts on making this more readable, feel free to suggest!

Reply:
I agree this makes it much more readable!